Document cacheSize option in usage docs (#642)
commercetools/commercetools-sync-java: docs/usage/PRODUCT_SYNC.md

         ProductSyncOptionsBuilder.of(sphereClient).batchSize(50).build();
````

##### cacheSize

> In the service classes of the commercetools-sync-java library, we have implemented an in-memory [LRU cache](https://en.wikipedia.org/wiki/Cache_replacement_policies#Least_recently_used_(LRU)) to store a map for commercetools reference key to reference ids.  This is used for the reference resolution of the library to reduce reference resolution based calls to the 

Reviewer: Do we need to tell whats stored in the map?
```suggestion
In the service classes of the commercetools-sync-java library, we have implemented an in-memory [LRU cache](https://en.wikipedia.org/wiki/Cache_replacement_policies#Least_recently_used_(LRU)) to store a map used for the reference resolution of the library.
```

> commercetools API, so the library will go only one time to fetch an id of resource, so the other references that are referencing the same resource might use the id in the cache instead of going to the commercetools API, 

Reviewer: Maybe shorten this sentence to make it easier to read. 
```suggestion
The cache reduces the reference resolution based calls to the commercetools API as the required fields of a resource will be fetched only one time. This cached fields then might be used by another resource referencing the already resolved resource instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve overall performance of the sync library and commercetools API.
```

> The cache reduces the reference resolution based calls to the commercetools API as the required fields of a resource will be fetched only one time. This cached fields then might be used by another resource referencing the already resolved resource instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve overall performance of the sync library and commercetools API.

Reviewer 2: ```suggestion
The cache reduces the reference resolution based calls to the commercetools API as the required fields of a resource will be fetched only one time. This cached fields then might be used by another resource referencing the already resolved resource instead of fetching from commercetools API. It turns out, having the in-memory LRU cache will improve overall performance of the sync library and commercetools API,
```
